staging/src/k8s.io/apiserver/pkg/server/options/feature.go

```diff
@@ -42,7 +42,10 @@ func NewFeatureOptions() *FeatureOptions {
 		EnableProfiling:           defaults.EnableProfiling,
 		DebugSocketPath:           defaults.DebugSocketPath,
 		EnableContentionProfiling: defaults.EnableContentionProfiling,
-		EnablePriorityAndFairness: true,
+		// Disable PriorityAndFairness by default since it only makes sense to have
+		// on by default for the kube-apiserver and not for the aggregated
+		// apiservers that are reusing this library.
```

Comment:
```
		// Disable PriorityAndFairness by default since it only makes sense to have
		// on by default for the kube-apiserver and not for the aggregated
		// apiservers that are reusing this library.
```

I don't agree with this comment

Reply:
it is equally important that system controllers get priority when talking to aggregated API servers. P&F operating at that server level as well helps ensure that.

Reply:
BTW, see related clarification in https://github.com/kubernetes/website/pull/46769.

@liggitt: My understanding is that the default and recommended configuration is that all clients go through the kube-apiserver. In this configuration it makes sense for APF to be disabled in the aggregated apiservers. I agree that it should be possible to operate in a configuration where clients access an aggregated apiserver directly and APF is enabled in that aggregated apiserver. However, in this situation you would want _all_ clients to access that aggregated server directly; two APF in series would not work well. This means, among other things, that the generic object garbage collector is not working on the objects in the aggregated server. And that means no APIService for the aggregate server, right? Does anybody actually do this? I see no reason to forbid it. But it is far from the common case. That is why I think it makes sense for the apiserver library to _default_ to disabling APF.

Reply:
> However, in this situation you would want all clients to access that aggregated server directly; [...] This means, among other things, that the generic object garbage collector is not working on the objects in the aggregated server. And that means no APIService for the aggregate server, right?

I don't see how any of those three things follow

Reply:
Let's take them one at a time. The first thing is what can be bad about two APF in series, first in kube-apiserver then nested in an aggregated apiserver. One potential issue with that is priority inversion: a delegated request might (get classified to a low priority and be queued for a while) at the aggregated apiserver while the original request at the kube-apiserver was classified as high priority. Another issue with that is more basic: if the delegated request is queued at the aggregated apiserver, then the original request at the kube-apiserver is waiting but it is not waiting in a queue managed by the APF in the kube-apiserver. So the first APF's management of the kube-apiserver's concurrency is getting confused. That is why things work better if there is only one APF in series.

Reply:
@liggitt: the generic object garbage collector works on all the resources defined through APIService objects, right? So to keep the generic object garbage collector away from objects stored in an external apiserver, the kube-apiserver has to have no APIService referring to their resource(s).